api/appbean/AppDetail.go

```diff
@@ -111,10 +112,10 @@ type CiPipelineDetails struct {
 }
 
 type CiPipelineMaterialConfig struct {
-	Type          pipelineConfig.SourceType `json:"type,omitempty" validate:"oneof=SOURCE_TYPE_BRANCH_FIXED SOURCE_TYPE_BRANCH_REGEX SOURCE_TYPE_TAG_ANY WEBHOOK"`
-	Value         string                    `json:"value,omitempty" `
-	CheckoutPath  string                    `json:"checkoutPath"`
-	GitMaterialId int                       `json:"gitMaterialId"`
+	Type          repository.SourceType `json:"type,omitempty" validate:"oneof=SOURCE_TYPE_BRANCH_FIXED SOURCE_TYPE_BRANCH_REGEX SOURCE_TYPE_TAG_ANY WEBHOOK"`
```

Comment:
as this is used in gitMaterial and gitHost, we should keep this at their common place instead of gitMaterial. This can cause issues in future